controllers/namespace/namespace_controller.go

```diff
@@ -83,23 +85,24 @@ func (r *ReconcileNamespaces) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, fmt.Errorf("failed to query Namespace: %w", err)
 	}
 
-	if ns.Labels == nil {
-		return reconcile.Result{}, nil
+	codeModuleDynakubes, err := codemodules.FindCodeModules(ctx, r.client)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to query CodeModule enabled Dynaekubes: %w", errors.WithStack(err))
 	}
 
-	oaName := ns.Labels[webhook.LabelInstance]
-	if oaName == "" {
-		return reconcile.Result{}, nil
+	dk, err := codemodules.MatchForNamespace(codeModuleDynakubes, &ns)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not match CodeModule with namespace: %w", errors.WithStack(err))
 	}
```

Comment:
Beware of excessive error logging and re-triggering the reconciler:
1. Does this justify an error message including a stack trace?
2. Will an immediate reconcile solve the problem or would it be sufficient to wait a couple more minutes?

Therefore, I suggest to log an info message and reschedule the event after a couple of minutes.

Reply:
changed